qt/scientific_interfaces/ISISReflectometry/Common/Parse.h

```diff
@@ -0,0 +1,61 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#ifndef MANTID_CUSTOMINTERFACES_PARSE_H_
+#define MANTID_CUSTOMINTERFACES_PARSE_H_
+#include "DllConfig.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/optional.hpp>
+#include <vector>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+MANTIDQT_ISISREFLECTOMETRY_DLL boost::optional<double>
+parseDouble(std::string string);
```

Comment:
Not sure why most of these are passing string by value? const ref would make more sense.